src/cascadia/TerminalApp/TabManagement.cpp

```diff
@@ -420,7 +420,7 @@ namespace winrt::TerminalApp::implementation
                     // GH#11356 - we can't use the UWP apis for writing the file,
                     // because they don't work elevated (shocker) So just use the
                     // shell32 file picker manually.
-                    path = co_await SaveFilePicker(*_hostingHwnd, [control](auto&& dialog) {
+                    path = co_await SaveFilePicker(*_hostingHwnd, [&tab](auto&& dialog) {

```

Comment:
We may actually want to capture `tab` by copy, if we can. It will increment a reference to the underlying smart pointer.

Capturing a reference with `&` during a coroutine handoff can in some cases result in the function blowing up later when it refers to the old coroutine's stack location. It can be messy.

Reply:
Copy constructor for `TerminalTab` is implicitly deleted cuz deep down `winrt::impl::root_implements` has `std::atomic` in it which is not CopyConstructible.

Reply:
*Oh*, I didn't realize it was a type from the `implementation` namespace. Sorry!

Reply:
Well, since all we need from tab is it's title we can get it outside of coroutine and just pass `wstring` by value.
```
std::wstring filename{ tab.Title() };
filename = til::clean_filename(filename);
path = co_await SaveFilePicker(*_hostingHwnd, [filename](auto&& dialog) {
```
Will this be fine?

Reply:
Ah, I love it! You can move the filename into the lambda as well!